pyrsistent/_pclass.py

```diff
@@ -139,6 +139,10 @@ def transform(self, *transformations):
         return transform(self, transformations)
 
     def __eq__(self, other):
+        if self is other:
+            return True
+        if hash(self) != hash(other):
```

Comment:
I think the `hash` comparison should come after the class inspection below. Otherwise we're not raising `NotImplementedError` when comparing with items of different types.
And I guess there must be missing test coverage for that case, in which case we could add some.